autopilot/contractor/state.go

```diff
@@ -26,11 +26,11 @@ type (
 
 	mCtx struct {
 		ctx   context.Context
-		state *MaintenanceState
+		state MaintenanceState
```

Comment:
While not wrong keeping the pointer here saves quite a bit of copying for a large struct like the state.

Reply:
I removed it because I assumed we _needed_ it at one point (due to a mutex or whatever) and forgot to clean this up. The state object is not a large struct imo. I reverted it to a pointer though, I don't really mind but in my head if I see a pointer I'm immediately assuming it _has_ to be a pointer and it's not there to avoid copying the struct.